xc7/fasm2bels/iob_models.py

```diff
@@ -126,21 +160,70 @@ def process_iob(top, iob):
     INTERMDISABLE_USED = site.has_feature('INTERMDISABLE.I')
     IBUFDISABLE_USED = site.has_feature('IBUFDISABLE.I')
 
-    top_wire = None
-    ilogic_active = False
-    ologic_active = False
+    # Collect all IOSTANDARD+DRIVE and IOSTANDARD+SLEW. Collect also possible
+    # input IOSTANDARDS.
+    iostd_drive = {}
+    iostd_slew = {}
+    iostd_in = set()
+
+    for feature in site.set_features:
+        parts = feature.split(".")
+
+        if "DRIVE" in parts:
+            idx = parts.index("DRIVE")
+
+            drives = [int(s[1:]) for s in parts[idx + 1].split("_")]
+            iostds = [s for s in parts[idx - 1].split("_")]
+
+            for ios in iostds:
+                if ios not in iostd_drive.keys():
+                    iostd_drive[ios] = set()
+                for drv in drives:
+                    iostd_drive[ios].add(drv)
+
+        if "SLEW" in parts:
+            idx = parts.index("SLEW")
+
+            slew = parts[idx + 1]
+            iostds = [s for s in parts[idx - 1].split("_")]
+
+            for ios in iostds:
+                if ios not in iostd_slew.keys():
+                    iostd_slew[ios] = slew
+
+        if "IN" in parts or "IN_ONLY" in parts:
+            iostd_in |= set([s for s in parts[-2].split("_")])
+
+    # Possible output configurations
+    iostd_out = []
+    for iostd in set(list(iostd_drive.keys())) | set(list(iostd_slew.keys())):
+        if iostd in iostd_drive and iostd in iostd_slew:
+            for drive in iostd_drive[iostd]:
+                iostd_out.append((
+                    iostd,
+                    drive,
+                    iostd_slew[iostd],
+                ))
+
+    # Buffer direction
+    is_input = has_feature_with_part(site, "IN") or has_feature_with_part(
+        site, "IN_ONLY"
+    )
+    is_inout = has_feature_with_part(site, "IN") and has_feature_with_part(
+        site, "DRIVE"
+    )
+    is_output = not has_feature_with_part(site, "IN") and \
+        has_feature_with_part(site, "DRIVE")
```

Comment:
Should sanity check this (e.g. only one is set).

Reply:
Done.